### DISCUSSION TITLE
geometry_area_perimeter returning 0 for global polygon

### DISCUSSION THREAD

**Discussion by jonathan-geoseer** [pyproj4/pyproj#1399] [Q&A]
I'm trying to work out the surface area of polygons on earth. It works fine for Colorado (conveniently square), but if I enter a global bounding box, I get a value of 0.
If I enter a sub-global one, I get a random-seeming negative number.

```
from shapely import box
from pyproj import Geod

geom = box(xmin=-180, ymin=-90, xmax=180, ymax=90)

# Sub global
# geom = box(xmin=-179.99, ymin=-89.99, xmax=179.99, ymax=89.99)

# Colorado
# geom = box(xmin=-109.05, ymin=37, xmax=-102.05, ymax=41)


geod = Geod(ellps="WGS84")

# Get the area and the perimeter
area, _ = geod.geometry_area_perimeter(geom)
print(area)
```

Results:
```
Global: 0
Colorado: 269154549884.00977  (Correct)
Sub-global: -28336978549.167137  (Huh????)

```

What's going on? There's nothing about these caveats in the docs as far as I can see: https://pyproj4.github.io/pyproj/stable/api/geod.html#pyproj.Geod.geometry_area_perimeter

Pyproj 3.5.0, Shapely 2.0.4

I can't decide if I'm missing something blindingly obvious, or this is a bug.

**Comment by snowman2** [comment]
See: https://github.com/pyproj4/pyproj/issues/1074#issuecomment-1124461148

**Reply by jonathan-geoseer** [reply]
Thanks.
That says:
> `geometry_area_perimeter` won't return more than half the area of the earth.

To those of us users who don't have any understanding of the maths, that makes no sense I'm afraid. And even if I did understand the maths, as a user I'd still expect a function called `geometry_area_perimeter` to work for arbitrarily large, globe-covering areas.

As such I'd suggest:
* This limitation needs to be explicitly documented. I'm evidently not the first person it has caught out.
* There should be a method/function for getting any sized area on the globe. It's certainly a regularly required feature: https://stackoverflow.com/questions/23697374/calculate-polygon-area-in-planar-units-e-g-square-meters-in-shapely and https://gis.stackexchange.com/questions/127607/calculating-area-in-km²-for-polygon-in-wkt-using-python


If there's some other way to do this in pyproj/shapely currently, it's well hidden: I've spent all morning trying to find a replacement after the old way was deprecated (just done an upgrade), so this would seem like a regression. In the end I've switched my area calculations to using the `area` library.

**Reply by snowman2** [reply]
Those are great suggestions. Contributions are welcome!

**Reply by jonathan-geoseer** [reply]
Thanks.

Documentation bug: https://github.com/pyproj4/pyproj/issues/1400
Polygon area calculation feature request: https://github.com/pyproj4/pyproj/issues/1401
